src/Build.OM.UnitTests/Construction/ProjectFormatting_Tests.cs

```diff
@@ -605,5 +603,86 @@ public void ChangeItemTypeWithXmlHeader()
 
             VerifyAssertLineByLine(expected, actual);
         }
+
+        [Fact]
+        public void VerifyUtf8WithoutBomTreatedAsUtf8()
+        {
+            string expected = ObjectModelHelpers.CleanupFileContents(@"<Project>
+</Project>");
+
+            Project project = new Project(NewProjectFileOptions.None);
+            StringWriter writer = new EncodingStringWriter(new UTF8Encoding(true));
+            project.Save(writer);
+
+            string actual = writer.ToString();
+
+            VerifyAssertLineByLine(expected, actual);
+        }
+
+        [Fact]
+        public void ProjectFileWithBomContainsBomAfterSave()
+        {
+            string content = ObjectModelHelpers.CleanupFileContents(@"<Project />");
+
+            var file = FileUtilities.GetTemporaryFile(".proj");
+            try
+            {
+                File.WriteAllText(file, content, new UTF8Encoding(true));
+                File.SetLastWriteTime(file, DateTime.Now.Subtract(TimeSpan.FromDays(1)));
+                Assert.True(FileStartsWithBom(file));
+
+                // Load and manipulate/save the project
+                var project = new Project(ProjectRootElement.Open(file, ProjectCollection.GlobalProjectCollection));
+                project.AddItem("Compile", "Program.cs");
+                project.Save();
+
+                // Ensure the file was really saved and that the file still contains a BOM
+                Assert.True(File.GetLastWriteTime(file).Subtract(DateTime.Now).TotalSeconds < 2);
+                Assert.True(File.ReadAllText(file).Contains("<Compile Include=\"Program.cs\" />"));
```

Comment:
`Assert.Contains(File.ReadAllText(file), "<Compile Include=\"Program.cs\" />")` ?

I think that'd give a nicer error if it failed.